plugins/catalog-react/src/apis/EntityPresentationApi/defaultEntityPresentation.ts

```diff
@@ -0,0 +1,197 @@
+/*
+ * Copyright 2023 The Backstage Authors
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+import {
+  CompoundEntityRef,
+  DEFAULT_NAMESPACE,
+  Entity,
+  stringifyEntityRef,
+} from '@backstage/catalog-model';
+import { IconComponent } from '@backstage/core-plugin-api';
+import ApartmentIcon from '@material-ui/icons/Apartment';
+import BusinessIcon from '@material-ui/icons/Business';
+import ExtensionIcon from '@material-ui/icons/Extension';
+import HelpIcon from '@material-ui/icons/Help';
+import LibraryAddIcon from '@material-ui/icons/LibraryAdd';
+import LocationOnIcon from '@material-ui/icons/LocationOn';
+import MemoryIcon from '@material-ui/icons/Memory';
+import PeopleIcon from '@material-ui/icons/People';
+import PersonIcon from '@material-ui/icons/Person';
+import get from 'lodash/get';
+import { EntityRefPresentationSnapshot } from './EntityPresentationApi';
+
+const UNKNOWN_KIND_ICON: IconComponent = HelpIcon;
+
+const DEFAULT_ICONS: Record<string, IconComponent> = {
+  api: ExtensionIcon,
+  component: MemoryIcon,
+  system: BusinessIcon,
+  domain: ApartmentIcon,
+  location: LocationOnIcon,
+  user: PersonIcon,
+  group: PeopleIcon,
+  template: LibraryAddIcon,
+};
+
+/**
+ * This returns the default representation of an entity.
+ *
+ * @public
+ * @param entityOrRef - Either an entity, or a ref to it.
+ * @param context - Contextual information that may affect the presentation.
+ */
+export function defaultEntityPresentation(
+  entityOrRef: Entity | CompoundEntityRef | string,
+  context?: {
+    defaultKind?: string;
+    defaultNamespace?: string;
+  },
+): EntityRefPresentationSnapshot {
+  // NOTE(freben): This code may look convoluted, but it tries its very best to
+  // be defensive and handling any type of malformed input and still producing
+  // some form of result without crashing.
+  const { kind, namespace, name, title, description, displayName, type } =
+    getParts(entityOrRef);
+
+  const Icon =
+    (kind && DEFAULT_ICONS[kind.toLocaleLowerCase('en-US')]) ||
+    UNKNOWN_KIND_ICON;
+
+  const entityRef: string = stringifyEntityRef({
+    kind: kind || 'unknown',
```

Comment:
Usage of `??` instead of `||` might be more appropriate

https://stackoverflow.com/questions/61480993/when-should-i-use-nullish-coalescing-vs-logical-or

Reply:
This code is special. The defaultEntityPresentation implementation needs to be so forgiving, that it can accept basically any malformed input while still producing sensible output and not crash while doing so. These actually want to be able to match on empty-string values.